skfda/preprocessing/registration/elastic.py

```diff
@@ -0,0 +1,780 @@
+    Fisher-Rao distance between functions, wich under the SRSF representation
+    becomes the usual :math:`\mathbb{L}^2` distance between functions.
+    See [SK16-4-6]_ .
+
+    The inverse SRSF transform is defined as
+
+    .. math::
+        f(t) = f(a) + \int_{a}^t q(t)|q(t)|dt .
+
+    This transformation is a mapping up to constant. Given the SRSF and the
+    initial value :math:`f(a)` the original function can be obtained, for this
+    reason it is necessary to store the value :math:`f(a)` during the fit,
+    which is dropped due to derivation. If it is applied the inverse
+    transformation without fit the estimator it is assumed that :math:`f(a)=0`.
+
+    Attributes:
+        eval_points (array_like, optional): Set of points where the
+            functions are evaluated, by default uses the sample points of the
+            fdatagrid.
+        store_initial (bool): If true stores the value :math:`f(a)` of the
+            samples during fitting to apply the inverse transform.
+            Defaults True.
+
+    Note:
+        Due to the use of derivatives it is recommended that the samples are
+        sufficiently smooth, or have passed a smoothing preprocessing before,
+        in order to achieve good results.
+
+    References:
+        ..  [SK16-4-6] Srivastava, Anuj & Klassen, Eric P. (2016). Functional
+            and shape data analysis. In *Square-Root Slope Function
+            Representation* (pp. 91-93). Springer.
+
+    Examples:
+
+        Create a toy dataset and apply the transformation and its inverse.
+
+        >>> from skfda.datasets import make_sinusoidal_process
+        >>> from skfda.preprocessing.registration.elastic import SRSF
+        >>> fd = make_sinusoidal_process(error_std=0, random_state=0)
+        >>> srsf = SRSF()
+        >>> srsf
+        SRSF(output_points=None, store_initial=True)
+
+        Fits the estimator (to apply the inverse transform) and apply the SRSF
+
+        >>> q = srsf.fit_transform(fd)
+
+        Apply the inverse transform.
+
+        >>> fd_pull_back = srsf.inverse_transform(q)
+
+        The original and the pull back `fd` are almost equal
+
+        >>> zero = fd - fd_pull_back
+        >>> zero.data_matrix.flatten().round(3)
+        array([ 0. ,  0. ,  0. , ... ])
+
+    """
+    def __init__(self, output_points=None, store_initial=True):
+        """Initializes the transformer.
+
+        Args:
+            eval_points: (array_like, optional): Set of points where the
+                functions are evaluated, by default uses the sample points of
+                the :class:`FDataGrid <skfda.FDataGrid>` transformed.
+            store_initial (bool): If true stores the value :math:`f(a)` of the
+                samples during fitting to apply the inverse transform.
+                Defaults True.
+
+        """
+        self.output_points = output_points
+        self.store_initial = store_initial
```

Comment:
In most cases, the inverse transformation is not used, and could be useful for performance reasons (in all internal calls related to the elastic alignment).

In addition, it is sometimes necessary to apply the inverse transformation of objects that have not been previously transformed, such as in the construction of the elastic mean, in which the mean of the initial values of the samples is taken as the initial value.

Anyway, I'm doubting whether the values should be saved when calling `transform` or `fit`.

Reply:
> In most cases, the inverse transformation is not used, and could be useful for performance reasons (in all internal calls related to the elastic alignment).

Ok.

> In addition, it is sometimes necessary to apply the inverse transformation of objects that have not been previously transformed, such as in the construction of the elastic mean, in which the mean of the initial values of the samples is taken as the initial value.

The way it is done now is a bit ugly. Maybe instead of a boolean you could set the value (and `None` would be the current `True` and 0 the current `False`).

> Anyway, I'm doubting whether the values should be saved when calling `transform` or `fit`.

These transformers are a bit weird, because they have to compute a value (initial value, warping function...) per sample. I do not think there is a precedent in scikit-learn, so we are free to choose. I think it makes more sense to do it in `transform` though, as it is part of the transformation.

Reply:
Maybe to change the convergence parameters `max_iter` or `tol`, although the default params will produce a good alignment in general.

Reply:
For those a lambda or partial suffices.

Reply:
> Maybe to change the convergence parameters max_iter or tol, although the default params will produce a good alignment in general.

Sorry, this message was for the other conversation, as an answer to:

> What kind of parameters are those you need to pass to the mean?

Reply:
> The way it is done now is a bit ugly. Maybe instead of a boolean you could set the value (and None would be the current True and 0 the current False).

> These transformers are a bit weird, because they have to compute a value (initial value, warping function...) per sample. I do not think there is a precedent in scikit-learn, so we are free to choose. I think it makes more sense to do it in transform though, as it is part of the transformation.

Done in 0abdca9.